### PR TITLE
ci: Github actions refactor

### PR DESCRIPTION
**Pull request by alexec**
Make builds faster and more robust.

### CODE REVIEW

File: .github/workflows/ci-gen.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+gen() {
+   sed "s/name: E2E/name: $1/" < e2e.yaml.0 |
+   sed "s/\${{matrix.test}}/$2/g" |
+   sed "s/\${{matrix.containerRuntimeExecutor}}/$3/g" |
+   ../../hack/auto-gen-msg.sh > $2-$3.yaml
+}
+
+gen "Test Docker Executor" test-executor docker
+gen "Test Emissary Executor" test-executor emissary
+gen "Test K8SAPI Executor" test-executor k8sapi
```
Comment (terrytangyuan) [inline .github/workflows/ci-gen.sh:22]: minor: K8SAPI -> K8sapi to be consistent with others (not all capitalized)

File: .github/workflows/test-executor-emissary.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# This is an auto-generated file. DO NOT EDIT
```
Comment (terrytangyuan) [inline .github/workflows/test-executor-emissary.yaml:1]: Add the source of these generated files?

Comment (alexec) [reply .github/workflows/test-executor-emissary.yaml:1]: `gen-ci.sh`

**Review by whynowy — APPROVED** [review]

**Comment by alexec** [comment]
Thank you Derek.

File: .gitignore [hunk 3]
```diff
@@ -39,3 +39,4 @@ git-ask-pass.sh
 pprof
 pkg/apiclient/sensor/sensor.swagger.json
 pkg/apiclient/eventsource/eventsource.swagger.json
+/argoexec-linux-amd64
```
Comment (simster7) [inline .gitignore:42]: @alexec Was this committed in error? If so, can I remove?

Comment (alexec) [reply .gitignore:42]: I've a follow-up PR to fix this: https://github.com/argoproj/argo-workflows/pull/5335 - you could approve that?

Comment (simster7) [reply .gitignore:42]: Done

**Pull request merged**